src/System.Security.AccessControl/tests/Ace.Tests.cs

```diff
@@ -0,0 +1,482 @@
+            Assert.Throws<ArgumentOutOfRangeException>("opaque", () => ObjectAce_CreateTestData(2, 1, 2, "S-1-5-11", 1, "{73D03E18-5C03-422c-9EC7-C4C5B1CB1612}", "{3CFD5DF9-9146-43c8-BF99-78E7E2DE4BAF}", true, 1, 0));
+            Assert.Throws<ArgumentOutOfRangeException>("opaque", () => ObjectAce_CreateTestData(2, 1, 2, "S-1-5-11", 1, "{73D03E18-5C03-422c-9EC7-C4C5B1CB1612}", "{3CFD5DF9-9146-43c8-BF99-78E7E2DE4BAF}", true, 17, 0));
+        }
+
+        [Fact]
+        public void ObjectAce_CreateBinaryForm_Invalid()
+        {
+            ObjectAce ace = (ObjectAce)ObjectAce_CreateTestData(0, 0, 1, "S-1-5-11", 1, "{73D03E18-5C03-422c-9EC7-C4C5B1CB1612}", "{3CFD5DF9-9146-43c8-BF99-78E7E2DE4BAF}", false, 8, 0)[0];
+            Assert.Throws<ArgumentNullException>("binaryForm", () => ObjectAce.CreateFromBinaryForm(null, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => ObjectAce.CreateFromBinaryForm(new byte[1], -1));
+            Assert.Throws<ArgumentException>("binaryForm", () => ObjectAce.CreateFromBinaryForm(new byte[ace.BinaryLength + 1], 2));
+            Assert.Throws<ArgumentException>("binaryForm", () => ObjectAce.CreateFromBinaryForm(new byte[ace.BinaryLength], 1));
+        }
+
+        [Fact]
+        public void ObjectAce_GetBinaryForm_Invalid()
+        {
+            ObjectAce ace = (ObjectAce)ObjectAce_CreateTestData(0, 0, 1, "S-1-5-11", 1, "{73D03E18-5C03-422c-9EC7-C4C5B1CB1612}", "{3CFD5DF9-9146-43c8-BF99-78E7E2DE4BAF}", false, 8, 0)[0];
+            Assert.Throws<ArgumentNullException>("binaryForm", () => ace.GetBinaryForm(null, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => ace.GetBinaryForm(new byte[1], -1));
+            Assert.Throws<ArgumentOutOfRangeException>("binaryForm", () => ace.GetBinaryForm(new byte[ace.BinaryLength + 1], 2));
+            Assert.Throws<ArgumentOutOfRangeException>("binaryForm", () => ace.GetBinaryForm(new byte[ace.BinaryLength], 1));
+        }
+
+        [Theory]
+        [MemberData(nameof(ObjectAce_TestObjects))]
+        public void ObjectAce_GetBinaryForm(GenericAce testAce, byte[] expectedBinaryForm, int testOffset)
+        {
+            byte[] resultBinaryForm = new byte[testAce.BinaryLength + testOffset];
+            testAce.GetBinaryForm(resultBinaryForm, testOffset);
+            GenericAce_VerifyBinaryForms(expectedBinaryForm, resultBinaryForm, testOffset);
+        }
+
+        [Theory]
+        [MemberData(nameof(ObjectAce_TestObjects))]
+        public void ObjectAce_CreateFromBinaryForm(GenericAce expectedAce, byte[] testBinaryForm, int testOffset)
+        {
+            GenericAce resultAce = ObjectAce.CreateFromBinaryForm(testBinaryForm, testOffset);
+            GenericAce_VerifyAces(expectedAce, resultAce);
+        }
+    }
+}
```

Comment:
If a test isn't in the PR it's because we didn't have a ToF equivalent. I won't be adding anything new in this PR, only porting what we already have, rewriting it to play nice with Xunit, and making sure there aren't any breaking failures.

That's also why there are a bunch of hard-coded values throughout the tests - because that's what was in the original tests. Like you, I don't know what most of the values mean; I'm just parroting them from ToF for the port. There are also a bunch of gaps in those tests, but that's out-of-scope for this push.

That in mind, I really should have made dotnet-bot author the initial commit and then added the playnice features on top of that. I'll rebase to clear that up.